src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,61 @@
+import classNames from 'classnames';
+import { useTodos } from '../../context';
+import { FilterType } from '../../types/FilterType';
+
+export const Footer = () => {
+  const { filter, setFilter, inProgress } = useTodos();
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${inProgress} ${inProgress <= 1 ? 'item' : 'items'} left`}
+      </span>
+
+      {/* Active filter should have a 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames('filter__link',
+            { selected: (filter === FilterType.All) })}
+          data-cy="FilterLinkAll"
+          onClick={() => setFilter(FilterType.All)}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={classNames('filter__link',
+            { selected: (filter === FilterType.Active) })}
+          data-cy="FilterLinkActive"
+          onClick={() => setFilter(FilterType.Active)}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={classNames('filter__link',
+            { selected: (filter === FilterType.Completed) })}
+          data-cy="FilterLinkCompleted"
+          onClick={() => setFilter(FilterType.Completed)}
+        >
+          Completed
+        </a>
```

Comment:
You can move the filter to another component and create an array to map filter values. In this scenario, if a value is repeated three times, you can express this more clearly

For Example

const FILTER_VALUES = ['All', 'Active', 'Completed'];
        
{FILTER_VALUES.map(filter => (
   <Filter filter={filter} key={filter} />
))}

Reply:
@kate-prakofyeva I approve this suggestion